### PR TITLE
GFORMS-1183 - Add a property to allow field labels to be sized as per…

### PR DESCRIPTION
**Pull request by SandipKhedekar**
… design system

### CODE REVIEW

File: app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala [hunk 1]
```diff
@@ -1958,6 +1981,17 @@ class SectionRenderingService(frontendAppConfig: FrontendAppConfig, lookupRegist
...
   }
 
+  private def getLabelClasses(isPageHeading: Boolean, labelSize: Option[LabelSize]): String =
+    (isPageHeading, labelSize) match {
+      case (_, Some(ExtraLarge)) => "govuk-label--xl"
+      case (_, Some(Large))      => "govuk-label--l"
+      case (_, Some(Medium))     => "govuk-label--m"
+      case (_, Some(Small))      => "govuk-label--s"
+      case (_, Some(ExtraSmall)) => "govuk-label--xs"
+      case (true, _)             => "govuk-label--l"
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/gform/SectionRenderingService.scala:1991]: If its page heading, we should default to large size as before right?

File: app/uk/gov/hmrc/gform/views/form/snippets/field_template_address.scala.html [hunk 2]
```diff
@@ -48,13 +49,13 @@
 <fieldset class="govuk-fieldset" id="@{fieldValue.id.value}-fieldset" role="group" @if(fieldValue.helpText.nonEmpty) { aria-describedby="@{fieldValue.id.value}-hint" }>
 
     @if(formLevelHeading) {
-        <legend class="govuk-fieldset__legend govuk-label--l">
+        <legend class="govuk-fieldset__legend govuk-label--l @classes">
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/views/form/snippets/field_template_address.scala.html:52]: Why do we need @classes if its formLevelHeading? it should be the default page level heading size i.e govuk-label--l right

File: app/uk/gov/hmrc/gform/views/form/snippets/field_template_overseas_address.scala.html [hunk 3]
```diff
@@ -52,13 +53,13 @@
 
 <fieldset class="govuk-fieldset" id="@{formComponent.id.value}-fieldset" role="group" @if(formComponent.helpText.nonEmpty) { aria-describedby="@{formComponent.id.value}-hint" }>
     @if(formLevelHeading) {
-        <legend class="govuk-fieldset__legend govuk-label--l">
+        <legend class="govuk-fieldset__legend govuk-label--l @classes">
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/views/form/snippets/field_template_overseas_address.scala.html:56]: same comment as before

File: app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html [hunk 4]
```diff
@@ -47,7 +48,7 @@
 <div class="gform-group">
 
   @if(label.nonEmpty) {
-    <label class="govuk-!-margin-bottom-7 govuk-label govuk-label--m">
+    <label class="govuk-!-margin-bottom-7 govuk-label govuk-label--m @classes">
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html:51]: Why have both govuk-label--m and @classes?

File: app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html [hunk 5]
```diff
@@ -45,11 +46,19 @@
...
 
 <div class="gform-group">
-
-  @if(label.nonEmpty) {
-    <label class="govuk-!-margin-bottom-7 govuk-label govuk-label--m">
-        @label
-    </label>
+  @(label.nonEmpty, classes.nonEmpty) match {
+    case (true, true) => {
+        alert(1)
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html:51]: what is this alert?

File: app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html [hunk 6]
```diff
@@ -45,11 +46,19 @@
...
+        <label class="govuk-!-margin-bottom-7 govuk-label @classes">
+            @label
+        </label>
+    }
+    case (true, false) => {
+        alert(2)
```
Comment (girishkamat) [inline app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html:57]: what is this alert?

File: app/uk/gov/hmrc/gform/views/form/snippets/field_template_overseas_address.scala.html [hunk 7]
```diff
@@ -24,6 +24,7 @@
     formComponent: FormComponent,
     validationResult: FormFieldValidationResult,
     formLevelHeading: Boolean,
+    classes: String,
```
[the remark this answers is not in the document]
Comment (SandipKhedekar) [reply app/uk/gov/hmrc/gform/views/form/snippets/field_template_overseas_address.scala.html:27]: **def getLabelClasses(_, _)** is returning String as **Label** and **Legend** in the library takes parameter **classes** as String. Hence I preferred String over Option.  And used Pattern Matching.

File: app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html (hunk 5 above)
Comment (SandipKhedekar) [reply app/uk/gov/hmrc/gform/views/form/snippets/group.scala.html:51]: Sorry. It was for testing.

File: app/uk/gov/hmrc/gform/views/form/snippets/field_template_overseas_address.scala.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (girishkamat) [reply app/uk/gov/hmrc/gform/views/form/snippets/field_template_overseas_address.scala.html:27]: It just creates a long list of case statements, which are bit annoying to read, especially in templates as the code editor does not support highlighting or syntax checks. Its preferable to keep the template as simple as possible. If its too much work, we can leave it this way.

**Review by girishkamat — APPROVED** [review]

**Pull request merged**